File: src/programmers/level2/위장.java
package programmers.level2;

import java.util.HashMap;
import java.util.Map;

public class 위장 {
    public static void main(String[] args) {
        String[][] clothes = {{"yellowhat", "headgear"}, {"bluesunglasses", "eyewear"}, {"green_turban", "headgear"}};
        solution(clothes);
    }

    public static int solution(String[][] clothes) {
        int combination = 1;
        Map<String, Integer> map = new HashMap<>();

        for (String[] str : clothes)
            map.put(str[1], map.getOrDefault(str[1], 0) + 1);

        for (String key : map.keySet())
            combination *= map.get(key) + 1;

        combination--;

        return map.size() == 1 ? clothes.length : combination;
    }
}

File: 프로그래머스/lv2/42578. 위장/위장.js
function solution(clothes) {
    const map = clothes.reduce((accu,curr)=> {
      accu.set(curr[1], (accu.get(curr[1])||0) +1) ;
      return accu;
    },new Map());
    
    let combination = 1;
    for(let [k, v] of map){
        combination *= v + 1;
    }
    
    return combination - 1;
}